charts/datadog/templates/otel-configmap.yaml

```diff
@@ -0,0 +1,65 @@
+{{- if eq (include "should-enable-otel-agent" .Values) "true" }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "agents-install-otel-configmap-name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{ include "datadog.labels" . | nindent 4 }}
+  annotations:
+    checksum/confd-config: {{ tpl (toYaml .Values.datadog.confd) . | sha256sum }}
+    checksum/autoconf-config: {{ tpl (toYaml .Values.datadog.autoconf) . | sha256sum }}
```

Comment:
it doesn't make sens that the checksum depends on the `confd-config` or `autoconf-config`. 

Instead it should be linked to the `.Values.datadog.otelCollector.config` or the default config. the idea is to force the checksum to change if the content of the configmap change.
Also this checksum should be use on top of the daemonset podtemplate to force a rollout of the agent if the otel config file has changed.

Reply:
see: https://github.com/DataDog/helm-charts/blob/d661a029e9dafb231f703dce5f579256d2e87f2f/charts/datadog/templates/daemonset.yaml#L47-L52